### ISSUE TITLE
Setup terraform state locking for namespaces in the environments repository

### ISSUE THREAD

**Issue by digitalronin** [ministryofjustice/cloud-platform#1334]
blocks #1115

When we implement #1115, we will have the current build pipeline applying all
namespaces in sequence, plus parallel pipelines applying individual namespaces
in response to merged PRs.

This means it's possible that both pipelines will try to update the same
namespace at the same time, which could lead to unpredictable results.

Before implementing #1115, we should add state-locking to the terraform states
in the environments repo. More details here:

https://www.terraform.io/docs/backends/types/s3.html

**Comment by digitalronin** [comment]
It looks like we can use a single dynamodb table to lock multiple different tfstate files. 

Initially, I think it makes sense to have one dynamodb table for the whole of the environments repository. Assuming this works out, I suggest we have a few more dynamodb tables - one each for:

* infrastructure/cloud-platform 
* infrastructure/cloud-platform-components
* concourse
* ...whatever I've forgotten

i.e. one table per "independent repository of terraform stuff"

**Comment by digitalronin** [comment]
This is working in the build pipeline. I broke the live-0 pipeline because the required dynamodb table wasn't there.

Creating the table in live-1 is done via the bootstrap pipeline, which applies the concourse terraform code where I defined the table. I couldn't figure out how to do the same for live-0, so I just created the same table manually in the moj-pi AWS account, using the web gui.

Live-0 is end of life, and if it ever died I think we'd just move everything to live-1 rather than try to recreate it, so I think this is acceptable.
